### PR TITLE
Examples of using X-Mailgun-Variables header

### PR DESCRIPTION
**Pull request by veulkehc**

### CODE REVIEW

File: source/samples/send-message-template-v.rst [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+         request.AddParameter ("v:body": "Sending messages with templates");
+         request.Method = Method.POST;
+         return client.Execute (request);
+     }
+
+ }
+
+.. code-block:: go
+
+ // Not implemented yet
```
Comment (thrawn01) [inline source/samples/send-message-template-v.rst]: Why is go always `// Not implemented yet`  

I have examples for every use case. I try to keep the golang client library up to date with every API mailgun has. If you don't think the library supports a feature please open an issue on the `mailgun/mailgun-go` repo and I'll get it done.

Here is the example that should be put in the `.. code-block:: go` 
https://github.com/mailgun/mailgun-go/blob/master/examples/examples.go#L900

/me cries in a corner.

File: source/user_manual.rst [hunk 2]
```diff
@@ -644,19 +644,11 @@ Just using the template name you can send a message:
 
 .. include:: samples/send-message-by-template-id.rst
 
-Another way to provide substitution variables is to use :ref:`X-Mailgun-Variables <manual-customdata>` in the http request i.e:
+Another way to provive variables is to use a form parameter prefixed with ``v:`` how it's showed :ref:`here <manual-customdata>`.
```
Comment (thrawn01) [inline source/user_manual.rst]: `provive` should be `provide`

File: source/samples/send-message-by-template-id.rst [hunk 3]
```diff
@@ -7,8 +7,7 @@
     -F to='alice@example.com' \
     -F subject='Hello' \
     -F template='template.test' \
-    -F v:title='API Documentation' \
-    -F v:body='Sending messages with templates'
+    -F h:X-Mailgun-Variabels='{"title": "API documentation", "body": "Sending messages with templates"}'
```
Comment (highlycaffeinated) [inline source/samples/send-message-by-template-id.rst]: s/b `Variables`

File: source/samples/send-message-by-template-id.rst [hunk 4]
```diff
@@ -27,13 +26,13 @@
...
- 	        .queryString("subject", "Hello")
-            .queryString("template", "template.test")
+            .field("from", "Excited User <YOU@YOUR_DOMAIN_NAME>")
+ 			.field("to", "alice@example.com")
+ 	        .field("subject", "Hello")
+            .field("template", "template.test")
  		    .field("o:tracking", "False")
-            .field("v:title", "API Documentation")
- 		    .asJson("v:body", "Sending messages with templates");
+            .field("h:X-Mailgun-Variabels", "{\"title\": \"API Documentation\", \"body\": \"Sending messages with templates\"}")
```
Comment (highlycaffeinated) [inline source/samples/send-message-by-template-id.rst]: s/b `Variables`

File: source/samples/send-message-by-template-id.rst [hunk 5]
```diff
@@ -51,12 +50,11 @@
...
-      'to'         => 'foo@example.com',
-      'subject'    => 'Hello',
-      'template'   => 'template.test',
-      'v:title'    => 'API Documentation',
-      'v:body'     => 'Sending messages with templates'
+      'from'                  => 'Excited User <YOU@YOUR_DOMAIN_NAME>',
+      'to'                    => 'foo@example.com',
+      'subject'               => 'Hello',
+      'template'              => 'template.test',
+      'h:X-Mailgun-Variabels' => '{"title": "API Documentation", "body": "Sending messages with templates"}'
```
Comment (highlycaffeinated) [inline source/samples/send-message-by-template-id.rst]: s/b `Variables`

File: source/user_manual.rst (hunk 2 above)
Comment (highlycaffeinated) [inline source/user_manual.rst]: s/showed/shown/

**Pull request merged**